vividus-bdd-engine/src/test/java/org/vividus/bdd/steps/VariableResolverTests.java

```diff
@@ -61,11 +61,13 @@ private Object convert(String value)
             " ${var:default},                       var:default,     value, value",
             " ${var:},                              var:,            value, value",
             " '{\"ids\": [\"${var}\", \"12-ce\"]}', var,             value, '{\"ids\": [\"value\", \"12-ce\"]}'",
-            " ${var${varPartName}},                 varPartName,     value, ${varvalue}"
+            " ${var${varPartName}},                 varPartName,     value, ${varvalue}",
+            " '#{exp(\\}{BNS_TRX_ID=, A, ${var})}', var,             value, '#{exp(\\}{BNS_TRX_ID=, A, value)}'",
+            " [JAVA_HOME=].*[^${PATH}].*,           var,             value, [JAVA_HOME=].*[^${PATH}].*"
     })
     void shouldResolveVariables(String valueToResolve, String variableKey, String variableValue, String expected)
     {
-        when(bddVariableContext.getVariable(variableKey)).thenReturn(variableValue);
+        lenient().when(bddVariableContext.getVariable(variableKey)).thenReturn(variableValue);
```

Comment:
why is `lenient` required now?

Reply:
Because there is an unnecessary mocking for the last example, but I feel it is better to keep all the cases in one place

Reply:
in this case the last value can be moved to `shouldReturnTheSameValueWhenNothingToResolve `